aiohttp/client.py

```diff
@@ -444,11 +444,7 @@ async def _request(
                     if proxy is not None:
                         proxy = URL(proxy)
                     elif self._trust_env:
-                        for scheme, proxy_info in proxies_from_env().items():
-                            if scheme == url.scheme:
-                                proxy = proxy_info.proxy
-                                proxy_auth = proxy_info.proxy_auth
-                                break
+                        proxy, proxy_auth = get_env_proxy_for_url(url)
```

Comment:
It looks like unconditionally re-assigning these variables changes the original behavior in that it overrides the values of `proxy` and `proxy_auth` provided as the arguments of `_request()`.
Before this change, the override only happened if anything was found. The helper function interface should be adjusted to communicate "not found" clearly and not substitute it with "found but None":
```suggestion
                        with contextlib.suppress(LookupError):
                            proxy, proxy_auth = get_env_proxy_for_url(url)
```